### PR TITLE
[Detector] rippling detector for phrase api tokens

### PR DESCRIPTION
**Pull request by SyedAliHamad**
### Description:
This PR adds a Detector for phase OAuth API token. 

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

**Comment by SyedAliHamad** [comment]
This PR introduces a new detector for Phrase Access Tokens.

**Detection Strategy:**

- **Keyword Filtering:** Efficiently scans for "phrase" keywords to identify relevant data chunks
- **Pattern Matching**: Uses regex to find 64-character hexadecimal tokens, typically prefixed with "phrase"
- **Deduplication:** Ensures each unique token is tested only once to prevent redundant API calls

Verification:
**Live API Testing:** Validates tokens against Phrase's` /v2/projects` endpoint. Uses the required Authorization: token <TOKEN> header format

**Response Handling:**
200 OK: Token is valid and active
401/403: Token is invalid or expired
Other statuses: Reported as verification errors

**Testing:**
Includes comprehensive tests covering pattern matching, live API verification with both active and inactive tokens and edge cases.

File: pkg/detectors/phraseaccesstoken/phraseaccesstoken.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		}
+
+		results = append(results, s1)
+	}
+
+	return results, nil
+}
+
+func verifyMatch(ctx context.Context, client *http.Client, token string) (bool, map[string]string, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://api.phrase.com/v2/projects", nil)
```
Comment (amanfcp) [inline pkg/detectors/phraseaccesstoken/phraseaccesstoken.go]:
```suggestion
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://api.phrase.com/v2/projects", http.NoBody)
```

File: pkg/detectors/phraseaccesstoken/phraseaccesstoken.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+			Raw:          []byte(token),
+		}
+
+		if verify {
+			client := s.client
+			if client == nil {
+				client = defaultClient
+			}
+
+			isVerified, extraData, verificationErr := verifyMatch(ctx, client, token)
```
Comment (amanfcp) [inline pkg/detectors/phraseaccesstoken/phraseaccesstoken.go]: No need to introduce extraData here if we're just returning nil.

**Review by amanfcp — CHANGES_REQUESTED** [review]

**Comment by shahzadhaider1** [comment]
I can't see `pkg/engine/defaults/defaults.go` being updated in this PR. Why is that?
Also, please resolve the conflicts.

File: pkg/detectors/phraseaccesstoken/phraseaccesstoken.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	for _, match := range matches {
+		// Ensure we have a proper capture group
+		if len(match) < 2 {
+			continue
+		}
+
+		token := match[1]
+		if token == "" {
+			continue
+		}
```
Comment (shahzadhaider1) [inline pkg/detectors/phraseaccesstoken/phraseaccesstoken.go]: why or when would a token be an empty string?

File: pkg/detectors/phraseaccesstoken/phraseaccesstoken_test.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	"github.com/google/go-cmp/cmp"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/engine/ahocorasick"
+)
+
+var (
+	validPattern   = "1a2b3c4d5e6f7890abcdef1234567890abcdef1234567890abcdef1234567890"
+	invalidPattern = "7cf4135a4e7f7ac228d36f210f151917a86f5dbd6"
+	keyword        = "phrase"
+)
```
Comment (shahzadhaider1) [inline pkg/detectors/phraseaccesstoken/phraseaccesstoken_test.go]: Can we use more than one valid pattern? 
Also, consider using the secret directly in the input text instead of assigning it to a variable; it'll improve readability.

Comment (shahzadhaider1) [reply pkg/detectors/phraseaccesstoken/phraseaccesstoken_test.go]: You can take a look at this PR for examples: https://github.com/trufflesecurity/trufflehog/pull/4359

**Review by shahzadhaider1 — APPROVED** [review]

**Review by amanfcp — APPROVED** [review]

File: pkg/detectors/phraseaccesstoken/phraseaccesstoken.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		token := match[1]
+
+		if !slices.Contains(tokens, token) {
+			tokens = append(tokens, token)
+		}
+	}
+
```
Comment (kashifkhan0771) [inline pkg/detectors/phraseaccesstoken/phraseaccesstoken.go]:
```suggestion
        var tokens = make(map[string]struct{})

	for _, match := range keyPat.FindAllStringSubmatch(dataStr, -1) {
		tokens[match[1]] = struct{}{}
	}
```

**Review by kashifkhan0771 — APPROVED** [review]
LGTM!
Good Work on pattern test cases ❤️ 

Just one small non-blocking suggestion

File: pkg/detectors/phraseaccesstoken/phraseaccesstoken.go (hunk 5 above)
Comment (shahzadhaider1) [reply pkg/detectors/phraseaccesstoken/phraseaccesstoken.go]: addressed
